hazelcast/src/test/java/com/hazelcast/query/SampleTestObjects.java

```diff
@@ -391,6 +394,86 @@ public String toString() {
         }
     }
 
+    // Mockito encounters non-deterministic failures in some tests when mocking
+    // the Employee object - to avoid further time spent investigating that issue,
+    // we simply provide our own basic spied invocation counter instead
+    public static class SpiedEmployee extends Employee {
+        private final Map<String, AtomicInteger> invocationCounters = new ConcurrentHashMap<>();
+
+        public SpiedEmployee(long id, String name, int age, boolean live, double salary, State state) {
+            super(id, name, age, live, salary, state);
+        }
+
+        public SpiedEmployee(long id, String name, int age, boolean live, double salary) {
+            super(id, name, age, live, salary);
+        }
+
+        public SpiedEmployee(String name, int age, boolean live, double salary) {
+            super(name, age, live, salary);
+        }
+
+        public SpiedEmployee(String name, String city, int age, boolean live, double salary) {
+            super(name, city, age, live, salary);
+        }
+
+        public SpiedEmployee(long id, String name, String city, int age, boolean live, double salary) {
+            super(id, name, city, age, live, salary);
+        }
+
+        public SpiedEmployee() {
+        }
+
+        public long getId() {
```

Comment:
Question: if two threads will call getId, there's a chance that the second change will override invocationCounters("getId"). Is it ever called from multiple threads? If no, then better to change whole map to HashMap

Reply:
I wrote the `SpiedEmployee` class to be fairly dynamic and usable in multiple tests, should we need it in the future - with that in mind, I opted to use a `ConcurrentHashMap` with `AtomicInteger` values to keep it thread-safe and ensure we don't encounter any race conditions or data loss when using this implementation (even in a multi-threaded environment). Due to this class only being used during test runs, the performance overhead from this should not be an issue.